core/src/main/java/nu/yona/server/goals/entities/Goal.java

```diff
@@ -58,6 +78,23 @@ public ZonedDateTime getCreationTime()
 		return creationTime;
 	}
 
+	public ZonedDateTime getEndTime()
```

Comment:
While writing this I realize that the 'add inactivity' algorithm will not work yet for historical goals. Please create a ticket that it should be updated to also process the historical goals (it could use the `previousGoal` relation to discover those now, or we could maintain a list of historical goals at the user).